Based on upstream r430

--- origsrc/indicator-applet-12.10.1/configure.ac	2012-09-19 11:17:54.000000000 -0500
+++ src/indicator-applet-12.10.1/configure.ac	2016-12-02 10:56:16.626594500 -0600
@@ -28,7 +28,7 @@ m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RU
 GTK_REQUIRED_VERSION=3.1
 INDICATOR_REQUIRED_VERSION=0.3.92
 INDICATOR_PKG=indicator3-0.4
-APPLET_PKG=libpanelapplet-4.0
+APPLET_PKG=libpanel-applet
 
 PKG_CHECK_MODULES(APPLET, gtk+-3.0 >= $GTK_REQUIRED_VERSION
                           x11
--- origsrc/indicator-applet-12.10.1/src/applet-main.c	2012-09-19 11:17:44.000000000 -0500
+++ src/indicator-applet-12.10.1/src/applet-main.c	2016-12-02 11:04:38.758238500 -0600
@@ -693,7 +693,8 @@ menubar_press (GtkWidget * widget,
 }
 
 static void
-about_cb (GtkAction *action G_GNUC_UNUSED,
+about_cb (GSimpleAction *action G_GNUC_UNUSED,
+          GVariant  *parameter G_GNUC_UNUSED,
           gpointer   data G_GNUC_UNUSED)
 {
   static const gchar *authors[] = {
@@ -873,15 +873,21 @@ static gboolean
 applet_fill_cb (PanelApplet * applet, const gchar * iid G_GNUC_UNUSED,
                 gpointer data G_GNUC_UNUSED)
 {
-  static const GtkActionEntry menu_actions[] = {
-    {"About", GTK_STOCK_ABOUT, N_("_About"), NULL, NULL, G_CALLBACK(about_cb)}
+  static const GActionEntry menu_actions[] = {
+    {"about", about_cb }
   };
-  static const gchar *menu_xml = "<menuitem name=\"About\" action=\"About\"/>";
+  static const gchar *menu_xml = ""
+    "<section>"
+    "<item>"
+    "<attribute name=\"label\" translatable=\"yes\">_About</attribute>"
+    "<attribute name=\"action\">indicator-applet.about</attribute>"
+    "</item>"
+    "</section>";
 
   static gboolean first_time = FALSE;
   GtkWidget *menubar;
   gint indicators_loaded = 0;
-  GtkActionGroup *action_group;
+  GSimpleActionGroup *action_group;
 
 #ifdef INDICATOR_APPLET_SESSION
   /* check if we are running stracciatella session */
@@ -922,12 +922,11 @@ applet_fill_cb (PanelApplet * applet, co
   gtk_container_set_border_width(GTK_CONTAINER (applet), 0);
   panel_applet_set_flags(applet, PANEL_APPLET_EXPAND_MINOR);
   menubar = gtk_menu_bar_new();
-  action_group = gtk_action_group_new ("Indicator Applet Actions");
-  gtk_action_group_set_translation_domain (action_group, GETTEXT_PACKAGE);
-  gtk_action_group_add_actions (action_group, menu_actions,
-                                G_N_ELEMENTS (menu_actions),
-                                menubar);
-  panel_applet_setup_menu(applet, menu_xml, action_group);
+  action_group = g_simple_action_group_new ();
+  g_action_map_add_action_entries (G_ACTION_MAP (action_group), menu_actions,
+                                   G_N_ELEMENTS (menu_actions), menubar);
+  panel_applet_setup_menu(applet, menu_xml, action_group, GETTEXT_PACKAGE);
+  gtk_widget_insert_action_group (GTK_WIDGET (applet), "indicator-applet", G_ACTION_GROUP (action_group));
   g_object_unref(action_group);
 #ifdef INDICATOR_APPLET
   atk_object_set_name (gtk_widget_get_accessible (GTK_WIDGET (applet)),
